### PR TITLE
Enable validating connections at the start of a request

### PR DESCRIPTION
**Pull request by RobbieClarken**
This PR wires up the Django `request_started` signal to verify existing connections are still usable by executing a simple database query. This behaviour is disabled by default and can be enabled by setting `VALIDATE_CONNECTION` to `True` in the database options.

The default query used to validate the connection is `SELECT 1 FROM sysmaster:sysdual` but can be overridden with the `VALIDATION_QUERY` option.

### CODE REVIEW

File: django_informixdb/base.py [hunk 1]
```diff
@@ -331,13 +351,25 @@ def _start_transaction_under_autocommit(self):
...
-            self.connection.cursor().execute("SELECT 1")
+            cursor = self.connection.cursor()
         except pyodbc.Error:
             return False
-        else:
-            return True
+
+        try:
+            cursor.execute(self._validation_query)
+        except pyodbc.Error:
```
Comment (richard-reece) [inline django_informixdb/base.py]: I think we should log the exception (just the text, as info).

File: django_informixdb/base.py [hunk 2]
```diff
@@ -331,13 +351,25 @@ def _start_transaction_under_autocommit(self):
...
+            return False
+        finally:
+            try:
+                cursor.close()
```
Comment (richard-reece) [inline django_informixdb/base.py]: I wonder whether it's worth mentioning in a comment *why* the code looks like this, to help a future dev avoid the pitfall of destructors not raising errors properly?

File: django_informixdb/base.py [hunk 3]
```diff
@@ -331,13 +351,25 @@ def _start_transaction_under_autocommit(self):
...
+            except pyodbc.Error:
```
Comment (richard-reece) [inline django_informixdb/base.py]: Also I think we should log the exception text here too (even though it's an "kinda expected" exception, the information could be valuable).

File: django_informixdb/base.py (hunk 2 above)
Comment (RobbieClarken) [reply django_informixdb/base.py]: There is a comment explaining this at the top of the function 😄

File: django_informixdb/base.py (hunk 3 above)
Comment (RobbieClarken) [reply django_informixdb/base.py]: 👍 Added.

File: django_informixdb/base.py (hunk 2 above)
Comment (RobbieClarken) [reply django_informixdb/base.py]: Added an extra comment because this whole `return` from within a `finally` is pretty weird.

Comment (richard-reece) [reply django_informixdb/base.py]: > There is a comment explaining this at the top of the function 😄

Oops 😅

**Pull request merged**